src/order.rs

```diff
@@ -0,0 +1,187 @@
+use std::cmp::Ordering;
+
+use async_graphql::{ComplexObject, Enum, Error, Result, SimpleObject};
+use bson::Uuid;
+use bson::{datetime::DateTime, Bson};
+use serde::{Deserialize, Serialize};
+
+use crate::foreign_types::UserAddress;
+use crate::order_datatypes::OrderDirection;
+use crate::order_item::OrderItemDTO;
+use crate::{
+    order_datatypes::CommonOrderInput, order_item::OrderItem,
+    order_item_connection::OrderItemConnection, user::User,
+};
+
+/// The Order of a user.
+#[derive(Debug, Serialize, Deserialize, PartialEq, Clone, SimpleObject)]
+#[graphql(complex)]
+pub struct Order {
+    /// Order UUID.
+    pub _id: Uuid,
+    /// User.
+    pub user: User,
+    /// Timestamp when Order was created.
+    pub created_at: DateTime,
+    /// The status of the Order.
+    pub order_status: OrderStatus,
+    /// Timestamp of Order placement. `None` until Order is placed.
+    pub placed_at: Option<DateTime>,
+    /// The rejection reason if status of the Order is `OrderStatus::Rejected`.
+    pub rejection_reason: Option<RejectionReason>,
+    /// The internal vector consisting of OrderItems.
+    #[graphql(skip)]
+    pub internal_order_items: Vec<OrderItem>,
+    /// Address to where the order should be shipped to.
+    pub shipment_address: UserAddress,
```

Comment:
the addresses are imho better located on Invoice/Shipment respectively, thus I would personally add a `#[graphql(skip)]` here, however, if you think we should keep them here too, we should adapt the domain model

Reply:
I have added a `#[graphql(skip)]` here, so not sure why Github is not showing this to me yet